### ISSUE TITLE
Deleting a federated repository does not delete the repo on remote members

### ISSUE THREAD

**Issue by pforristal-jll** [jfrog/terraform-provider-artifactory#704]
**Describe the bug**
Deleting a federated repository does not delete the repo on remote members.

```
terraform {
  required_version = "~>1.3"
  required_providers {
    artifactory = {
      source  = "jfrog/artifactory"
      version = "~> 7.2"
    }
  }
}

provider "artifactory" {
  # Configuration options
  url          = "https://${var.artifactory_domain_primary}/artifactory"
  access_token = var.artifactory_access_token
}

provider "artifactory" {
  # Configuration options
  alias        = "secondary"
  url          = "https://${var.artifactory_domain_secondary}/artifactory"
  access_token = var.artifactory_access_token
}


locals {
  environments       = ["dev", "prod"]
  npm_teams          = [ "team1","team2"]
  npm_team_repos = distinct(
    flatten(
      [
        for env in local.environments : [
          for team in local.npm_teams : {
            team = team
            env  = env
          }
        ]
      ]
    )
  )
}


variable "artifactory_domain_primary" {
  type        = string
  description = "Artifactory Primary Domain"
  default     = "primary.jfrog.io"
}

variable "artifactory_domain_secondary" {
  type        = string
  description = "Artifactory Secondary Domain"
  default     = "secondary.jfrog.io"
}

variable "artifactory_access_token" {
  sensitive   = true
  type        = string
  description = "Artifactory Access Token"
}

resource "artifactory_federated_npm_repository" "federated" {
  for_each = { for entry in local.npm_team_repos : "${entry.team}.${entry.env}" => entry }

  key = "${each.value.team}-npm-${each.value.env}-local"
  member {
    url     = "https://${var.artifactory_domain_primary}/artifactory/${each.value.team}-npm-${each.value.env}-local"
    enabled = true
  }

  member {
    url     = "https://${var.artifactory_domain_secondary}/artifactory/${each.value.team}-npm-${each.value.env}-local"
    enabled = true
  }
}
```

Create the repos, then run a terraform destroy.  the remote repository doesn't clean up the repos.

Artifactory Version: 7.55.6 (jfrog cloud)
Terraform Version: 1.4.2
Terraform Provider: 7.3.1

**Requirements for and issue**
- [x] A description of the bug
- [x] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue). **If this is not supplied, this issue will likely be closed without any effort expended.**
- [x] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [x] Your version of terraform
- [x] Your version of terraform provider

**Expected behavior**
When destroying resources it is expected for the resources to actually be destroyed.

**Additional context**
The documentation around the Federated repositories is quite thin, and leaves a lot to be desired, and some of the errors encountered are non-intuitive (e.g. if you follow the docs and forget to change `http://` to `https://`, you'll get a 302 redirect that breaks the provider.)

**Comment by danielmkn** [comment]
Hello @pforristal-jll! The member(s) of the federated repository is not getting deleted, when you delete the initial repository [per official documentation](https://jfrog.com/help/r/jfrog-artifactory-documentation/working-with-federated-repositories).

> **Removing Repositories from the Federation**
> Admins on any of the federated members can remove themselves and other members from the Federation. Note that you can delete the initial repository and the Federation will continue to function between the remaining federated members.

In our testing, I can see, that the repository, created on the second Artifactory instance stays after the initial repository was deleted. Both UI and API have the same behavior.

**Comment by pforristal-jll** [comment]
That's not really how terraform expect the resources to be handled (delete _should mean_ delete, not orphan), and leads to a lot of administrative overhead to cleanup.  Is there a supported way to use terraform to provision _and_ deprovision federated resources?

**Comment by danielmkn** [comment]
Yes, you're right. Our approach is to always mimic Artifactory UI/API functionality in our providers. And the product doesn't kill the members.
Theoretically, we can add functionality to "manually" delete all the members, if the initial repository is deleted. But I'm not sure if we should do this and if it won't be an unexpected surprise for existing users. We will discuss it internally first, then we can create a discussion on GitHub.

**Comment by pforristal-jll** [comment]
The bigger issue comes about if you delete and go to re-add a federated repository, it will halt the `terraform apply` because it runs into existing resources. That it doesn't know how to handle.

**Comment by danielmkn** [comment]
Yes, it will error out with 400 from the API. We will think about how to deal with that. I'll update here tomorrow, hopefully.

**Comment by pforristal-jll** [comment]
Making it a parameter like `cleanup_on_delete = true` that defaults to `false` would make it a non-breaking change, that adds the required functionality.

**Comment by danielmkn** [comment]
We will implement the cleanup function by removing the members from the federation before we delete the initial repository. This way, the repos on the remote instance won't be removed (the same way as Artifactory works), but users won't have a conflict when re-applying the configuration.

**Comment by pforristal-jll** [comment]
That still leaves orphaned resources that fall outside of the TF state/expectations and require manual intervention.

**Comment by danielmkn** [comment]
@pforristal-jll, we will delete all the members, if `cleanup_on_delete = true`, as you asked. It will be released in v7.5.0.

**Comment by danielmkn** [comment]
@pforristal-jll, it's released in v7.5.0
